### ISSUE TITLE
 Unable to get GCM project number even after setting gcm_key

### ISSUE THREAD

**Issue by kaushikkapil** [driftyco/ionic-service-push-client#23]
Hi @ericb 
I am facing an issue very similar to Issue https://github.com/driftyco/ionic-service-push-client/issues/21

I have set the "gcm_key" to a string value as you suggested above, but I still keep getting the error. If I open the ".io-config.json", I see the following entry:
    {"gcm_key":"1001628164286"}

Here below is the "live reload" log as you can see.

    1     124392   log      Ionic Core: searching for cordova.js
    2     124393   log      Ionic Core: cordova.js has already been loaded
    3     124597   warn     PUSH: Unable to get GCM project number, run "ionic config set gcm_key your-gcm-id"
    9     124720   log      Ionic Analytics: enqueuing event to send later:, Load, [object Object]
    10    124726   log      Ionic Analytics: enqueuing event to send later:, State Change, [object Object]
    11    124770   error    No Content-Security-Policy meta tag found. Please add one when using the cordova-plugin-whitelist plugin.
    26    125481   log      Ionic Analytics: event recieved but not sent (dryRun active):, State Change, [object Object]
    27    126267   log      Ionic Analytics: successfully registered analytics key
    28    126367   error    $ionicPush:REGISTER_ERROR, org.json.JSONObject$1 cannot be cast to java.lang.String
I have updated my ionic cli to the latest version as well, and followed the steps mentioned in the link below:
http://docs.ionic.io/docs/push-android-setup

Regards,
Kapil

**Comment by ericb** [comment]
@kaushikkapil I was not able to replicate this. I would check your ionic-service-core/ionic-core.js file and make sure the settings have correctly written there. You could also simply try running `ionic config build` and then `ionic prepare android` before running a build.

**Comment by kaushikkapil** [comment]
@ericb I was able to finally resolve this, by updating the ionic cli, and then running ionic config build. 

Though I was able to get notifications working, and push a notification successfully do the ionic app, it stopped working when I uninstalled and installed the app and tried to register again. The app stops working immediately after I call the register push function. The user gets registered on server, I do not see the Android Device id on the server. Not sure how to go about debugging it as it is not a javascript crash, but looks more like a phonegap level crash.
